lib/ex_machina/ecto.ex

```diff
@@ -0,0 +1,58 @@
+defmodule ExMachina.Ecto do
+  defmacro __using__(opts) do
+    quote do
+      use ExMachina
+
+      @repo Dict.get(unquote(opts), :repo)
```

Comment:
If you use Dict.fetch! It should crash at compile time, but I don't know of a way to get a nice error message. You could try get the option before the `quote` and raising there, but I've never done that